kore/src/Kore/Log/DebugTransition.hs

```diff
@@ -0,0 +1,117 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+-}
+module Kore.Log.DebugTransition (
+    DebugTransition (..),
+    debugBeforeTransition,
+    debugAfterTransition,
+    debugFinalTransition,
+) where
+
+import Kore.Attribute.Axiom (
+    SourceLocation,
+ )
+import Kore.Reachability.ClaimState (
+    ClaimState,
+ )
+import Kore.Reachability.Prim (
+    Prim (..),
+ )
+import Kore.Reachability.SomeClaim (
+    SomeClaim (..),
+ )
+import Log
+import Prelude.Kore
+import Pretty (
+    Pretty (..),
+ )
+import qualified Pretty
+
+data DebugTransition
+    = DebugAfterTransition
+        (Maybe (ClaimState SomeClaim))
+        Prim
+        [SourceLocation]
+    | DebugBeforeTransition
+        (ClaimState SomeClaim)
+        Prim
+    deriving stock (Show)
```

Comment:
Since the two constructors have more than one field, I think it would be best to create two record types with those fields, so in the end it will look like:

```suggestion
data DebugTransition
    = DebugAfterTransition AfterTransition
    | DebugBeforeTransition BeforeTransition
    deriving stock (Show)
```